pandas/tseries/index.py

```diff
@@ -731,19 +731,39 @@ def _add_datelike(self, other):
     def _sub_datelike(self, other):
         # subtract a datetime from myself, yielding a TimedeltaIndex
         from pandas import TimedeltaIndex
-        other = Timestamp(other)
-        if other is tslib.NaT:
-            result = self._nat_new(box=False)
-        # require tz compat
-        elif not self._has_same_tz(other):
-            raise TypeError("Timestamp subtraction must have the same "
-                            "timezones or no timezones")
+        if isinstance(other, DatetimeIndex):
```

Comment:
I moved the condition to base, but left it here as well (to follow the same pattern as for the other ops: one method in base (eg `_add_delta`) which in the subclass handles both scalars as index)